### PR TITLE
Clap cleanup

### PR DESCRIPTION
**Pull request by tertsdiepraam**
Hi! I took some time to clean up the clap parser implementation. The most important change is that I changed the `u8`s to `bool`s and removed `ArgAction::Count`, because that felt a bit non-sensical. Other cleanups:
- Change the type of the help field
- Use a derive macro for `Default` on `Opts`
- Remove explicit names for long options
- Remove a bunch on unused types and methods (i.e. some that were only used in tests or not at all)
- Clean up the `ValueEnum` implementations a bit.

They can all be reviewed in separate commits and I can take commits out if you want.

### CODE REVIEW

File: src/options/parser.rs
Comment (MartinFillon) [inline src/options/parser.rs:153]: This must be put in the documentation too.

**Review by MartinFillon — CHANGES_REQUESTED** [review]
Hey, Thanks for the heads up, for your information, the use of `u8` is justified
This is to prevent cases such as `ls -l` where `alias ls=eza --long` to fail, please revert this part of your commit. As this is an important feature to have and not to break.

If you want the full convo about this please see the closed clap prs

**Comment by tertsdiepraam** [comment]
Thanks for checking it out!

> If you want the full convo about this please see the closed clap prs

This convo or is there more context somewhere? https://github.com/eza-community/eza/pull/640#discussion_r1405381972

I'd still suggest using `args_override_self` instead of the `u8`s. I'll let the maintainers weigh in on this and I'll follow their decision. I'll also add tests for this behaviour regardless of the implementation we decide on.

**Comment by PThorpe92** [comment]
> Hey, Thanks for the heads up, for your information, the use of `u8` is justified This is to prevent cases such as `ls -l` where `alias ls=eza --long` to fail, please revert this part of your commit. As this is an important feature to have and not to break.
> 
> If you want the full convo about this please see the closed clap prs

Ok so lets have a convo about it. So you're saying the reason it was u8 instead of bool, was because if it gets passed twice, such as when it's already `--long` in the users alias, and again on the command line. 

@tertsdiepraam  was that behavior something a boolean flag could deal with/was that accounted for? I briefly remember it being brought up in the original PR

**Comment by MartinFillon** [comment]
> I'd still suggest using args_override_self instead of the u8s. 

Seems perfect tbh

**Comment by tertsdiepraam** [comment]
Awesome, I'll try to implement it!

**Review by MartinFillon — APPROVED** [review]
Thanks for the tests, and the documentation, LGTM.

**Pull request merged**
